src/librustc_errors/lib.rs

```diff
@@ -816,6 +827,16 @@ impl HandlerInner {
         }
     }
 
+    fn print_warning_count(&mut self) {
+        self.emit_stashed_diagnostics();
+        let s = match self.deduplicated_warn_count {
+            0 => return,
+            1 => "build completed with one warning".to_string(),
+            count => format!("build completed with {} warnings", count),
+        };
+        let _ = self.emit_diagnostic(&Diagnostic::new(Level::Warning, &s));
```

Comment:
Can't this logic go inside of `print_error_count`?

We could even get fancy and emit `aborting due to XX previous errors, YY warnings emitted`.

Reply:
Agreed, that could work.

Reply:
What would be your preferred way to print it if the build succeeds, but warnings are emitted?
I was thinking about leaving it as it is (`warning: build completed with x warnings` or something similar).